File: ecos-records/src/main/java/ru/citeck/ecos/records3/record/op/atts/service/value/factory/BooleanValueFactory.java
package ru.citeck.ecos.records3.record.op.atts.service.value.factory;

import ru.citeck.ecos.records3.record.op.atts.service.value.AttValue;

import java.util.Collections;
import java.util.List;

public class BooleanValueFactory implements AttValueFactory<Boolean> {

    @Override
    public AttValue getValue(Boolean value) {
        return new AttValue() {
            @Override
            public String asText() {
                return value.toString();
            }

            @Override
            public Boolean asBool() {
                return value;
            }
        };
    }

    @Override
    public List<Class<? extends Boolean>> getValueTypes() {
        return Collections.singletonList(Boolean.class);
    }
}

File: ecos-records/src/main/java/ru/citeck/ecos/records3/record/op/atts/service/value/factory/BooleanValueFactory.kt
package ru.citeck.ecos.records3.record.op.atts.service.value.factory

import ru.citeck.ecos.records3.record.op.atts.service.value.AttValue

class BooleanValueFactory : AttValueFactory<Boolean> {

    override fun getValue(value: Boolean): AttValue? {

        return object : AttValue {

            override fun asText(): String? {
                return value.toString()
            }

            override fun asBoolean(): Boolean? {
                return value
            }
        }
    }

    override fun getValueTypes() = listOf(Boolean::class.java)
}
